query/time_bucket_result.go

```diff
@@ -0,0 +1,102 @@
+package query
+
+import (
+	"fmt"
+
+	"github.com/xichen2020/eventdb/document/field"
+)
+
+// TimeBucketResults is a collection of time buckets recording the counts
+// for documents falling in each bucket.
+type TimeBucketResults struct {
+	StartBucketNanos int64 // Inclusive
+	BucketSizeNanos  int64
+	NumBuckets       int
```

Comment:
nit: we should be able to just use `len(buckets)` instead of having another attr.

Reply:
Also don't see where `NumBuckets` gets assigned a value.

Reply:
So It's assigned in `query.go`, the reason why we have `NumBuckets` is so that the buckets are allocated lazily.